build.gradle

```diff
@@ -78,6 +79,8 @@ dependencies {
 
     testCompile libraries.assertj
 
+    testCompile 'de.bechte.junit:junit-hierarchicalcontextrunner:4.12.2-SNAPSHOT'
```

Comment:
We don't add additional dependencies. Could you instead extract a minimal reproducible test in our repository which is based on your runner?

Reply:
That's more than fair.  At that time, I couldn't think of a way to reproduce this issue without including that dependency other than lifting the test from #353.  Ironically that dependency introduced the runner I was using to reproduce the issue.

I have force-updated my branch to remove the test dependency since I thought of a way to demonstrate the issue without it.  My old test (archived [here](https://github.com/mouyang/mockito/commit/c7e32b85abc401e54cb7c7a1d9c951db5d06fe5a)) was meant to verify an inner class could override mocks set in an outer class, but that is unnecessary on second thought.